src/app/donation-start/donation-start.component.ts

```diff
@@ -337,21 +337,22 @@ export class DonationStartComponent implements OnInit {
     this.charityCheckoutService.startDonation(donation);
   }
 
-  private promptToContinueWithNoMatchingLeft(donation: Donation) {
+  private promptToContinueWithNoMatchingLeft(donation: Donation, surplusDonationInfo?: string) {
```

Comment:
This method is already non-static and the campaign never changes. This shouldn't need a new input param? This should also make all changes in the above `.subscribe(...)` unnecessary.